2025年3月納品分差分反映

wbkataoka opened the pull request.

wbkataoka committed:
2025年3月納品分差分反映

AyanaOzeki commented:
the revisions are as follows.

Add buyer_data.user_id
- Set SFCC member ID
- Set null for guests

*Scope of modification
- SFRA and SiteGenesis
- standard and token payments

Fix to pass `$ npm run lint` and `$ npm run test`

And if there is no particular problem with this update, could you please publish it?

Sai-LaoKham-Paidy commented on package.json:
```diff
@@ -1,14 +1,13 @@
   "scripts": {
     "test": "sgmf-scripts --test test/unit/**/*.js",
-    "test:ci": "TZ=Asia/Tokyo npm run cover --ci --maxWorkers=3",
     "cover": "sgmf-scripts --cover 'test/unit'",
```
Removing this "test:ci" script led to this pipeline [failure](https://app.circleci.com/pipelines/github/paidy/SalesforceCommerceCloud-AppExchange/60/workflows/1bf7ce38-030d-4efb-9e83-2ec7fce97fb2/jobs/111). May I know why it's required to remove?

Sai-LaoKham-Paidy replied:
And if it's necessary to remove here, circleci config should also be updated.

AyanaOzeki replied:
@wbkataoka 以下の部分が削除されていたようですが、こちらは今回の対応にて必要な修正でしょうか。
`    "test:ci": "TZ=Asia/Tokyo npm run cover --ci --maxWorkers=3"`

wbkataoka replied:
@AyanaOzeki 
修正しました。

- Restored the deleted test:ci script in the scripts section.
- Excluded the folder generated by npm run test:ci from both Git and linting.

Sai-LaoKham-Paidy commented on package.json:
```diff
@@ -1,14 +1,13 @@
   "main": "index.js",
   "engines": {
-    "node": ">=10.24.1"
+    "node": ">=4.0"
```
I believe you should also update the node version [here](https://github.com/paidy/SalesforceCommerceCloud-AppExchange/blob/master/.circleci/config.yml#L22) in order to execute the pipeline successfully.

AyanaOzeki replied:
@wbkataoka  `"node": ">=4.0"`は従来の`"node": ">=10.24.1"`から更新となる想定ですが、4.0が正しいでしょうか。

wbkataoka replied:
@AyanaOzeki 
"node": ">=10.24.1" に戻して動作確認を行い、lint、test、test:ci のいずれも問題がなく動作することが確認できましたので、変更箇所を復元しました。

- Restored the required node version to >=10.24.1 in the engines field of package.json.

wbkataoka committed:
fix(package.json): revert unintended differences

- Restored the required node version to >=10.24.1 in the engines field of package.json.
-  Restored the deleted test:ci script in the scripts section.
- Excluded the folder generated by npm run test:ci from both Git and linting.
